#! /bin/sh -e
## 12_codec_gsm.c.dpatch by Kilian Krause <kk@verfaction.de> as was proposed by Thiemo Seufer
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: As we use the system libgsm, removed call to old function.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
--- asterisk-0.9.1+1.0RC1/codecs/codec_gsm.c.old	2004-08-17 09:03:27.000000000 +0200
+++ asterisk-0.9.1+1.0RC1/codecs/codec_gsm.c	2004-08-17 09:04:33.000000000 +0200
@@ -224,7 +224,7 @@
 static void gsm_destroy_stuff(struct ast_translator_pvt *pvt)
 {
 	if (pvt->gsm)
-		gsm_destroy(pvt->gsm);
+		free(pvt->gsm);
 	free(pvt);
 	localusecnt--;
 }
